plans/phase-1/api-design/multigres-operator-api-v1alpha1-design.md

```diff
@@ -0,0 +1,1226 @@
+  storage:
+    size: "10Gi"
+    class: "standard-gp3"
+  image: "quay.io/coreos/etcd:v3.5"
+  resources:
+    requests:
+      cpu: "500m"
+      memory: "1Gi"
+    limits:
+      cpu: "1"
+      memory: "2Gi"
+status:
+  conditions:
+    - type: Available
+      status: "True"
+      lastTransitionTime: "2025-11-07T12:01:00Z"
+  clientService: "example-cluster-global-client"
+  peerService: "example-cluster-global-peer"
+```
+
+#### Child CR: Cell
+
+  * Owned by `MultigresCluster`.
+  * Strictly contains `MultiGateway` and optional `LocalTopoServer`.
+  * The `allCells` field is used for discovery by gateways.
+
+```yaml
+apiVersion: multigres.com/v1alpha1
+kind: Cell
+metadata:
+  name: "example-cluster-us-east-1a"
+  namespace: example
+  labels:
+    multigres.com/cluster: "example-cluster"
+    multigres.com/cell: "us-east-1a"
+  ownerReferences:
+    - apiVersion: multigres.com/v1alpha1
+      kind: MultigresCluster
+      name: "example-cluster"
+      uid: "a1b2c3d4-1234-5678-90ab-f0e1d2c3b4a5"
+      controller: true
+spec:
+  name: "us-east-1a"
+  zone: "us-east-1a"
+
+  # Images passed down from global configuration
+  images:
+    multigateway: "multigres/multigres:latest"
+
+  # Resolved from CellTemplate + Overrides
+  multigateway:
+    replicas: 3
+    resources:
+      requests:
+        cpu: "500m"
+        memory: "512Mi"
+      limits:
+        cpu: "1"
+        memory: "1Gi"
+  
+  # A reference to the GLOBAL TopoServer.
+  # Always populated by the parent controller if no local server is used.
+  globalTopoServer:
+    address: "example-cluster-global-client.example.svc.cluster.local:2379"
+    rootPath: "/multigres/global"
+    implementation: "etcd2"
+
+  # Option 1: Using the Global TopoServer (Default)
+  #
+  topoServer: {}
+
+  # Option 2: Inline Definition (External)
+  #
+  # topoServer:
+  #   external:
+  #     address: "my-etcd.some-namespace.svc.cluster.local:2379"
+  #     rootPath: "/multigres/us-east-1a"
+  #     implementation: "etcd2"
+
+  # Option 3: Managed Local
+  #
+  # topoServer:
+  #   etcd:
+  #     rootPath: "/multigres/us-east-1a"
+  #     image: "quay.io/coreos/etcd:v3.5"
+  #     replicas: 3
+  #     storage:
+  #       size: "5Gi"
+  #       class: "standard-gp3"
+
+  # List of all cells in the cluster for discovery.
+  allCells:
+  - "us-east-1a"
+  - "us-east-1b"
+
+  # Topology flags for the Cell controller to act on.
+  topologyReconciliation:
+    registerCell: true
+    pruneTablets: true
+
+status:
+  conditions:
+  - type: Available
+    status: "True"
+    lastTransitionTime: "2025-11-07T12:01:00Z"
+  gatewayReplicas: 3
+  gatewayReadyReplicas: 3
+  gatewayServiceName: "example-cluster-us-east-1a-gateway"
+```
+
+#### Child CR: TableGroup
+
+  * Owned by `MultigresCluster`.
+  * Acts as the middle-manager for Shards. It MUST contain the full resolved specification for all shards it manages, enabling it to be the single source of truth for creating its child `Shard` CRs.
```

Comment:
I think grammatically this would be more correct?

```suggestion
  * Acts as the middle-manager for Shards. It MUST contain the fully resolved specification for all shards it manages, enabling it to be the single source of truth for creating its child `Shard` CRs.
```